### PR TITLE
Fix null-removal on nested captured-arrays

### PR DESCRIPTION
**Pull request by staabm**
as described in https://github.com/composer/pcre/pull/34#issue-2474070032

### CODE REVIEW

File: src/PHPStan/PregReplaceCallbackClosureTypeExtension.php [hunk 1]
```diff
@@ -61,6 +61,18 @@ public function getTypeFromStaticMethodCall(MethodReflection $methodReflection,
             $matchesType = new ConstantArrayType(
                 $matchesType->getKeyTypes(),
                 array_map(static function (Type $valueType): Type {
+                    if (count($valueType->getConstantArrays()) === 1) {
```
Comment (staabm) [inline src/PHPStan/PregReplaceCallbackClosureTypeExtension.php:64]: when `PREG_OFFSET_CAPTURE` the `$valueType` is again a constant-array-type. 

`TypeCombinator::removeNull` does not deep iterate into arrays

Comment (staabm) [reply src/PHPStan/PregReplaceCallbackClosureTypeExtension.php:64]: I used the same logic already arround, but I guess we will get problems in the future with it when patterns are used which return more then 1 constant array (maybe one of the tagged union cases)

Comment (staabm) [reply src/PHPStan/PregReplaceCallbackClosureTypeExtension.php:64]: we could use https://github.com/phpstan/phpstan-src/blob/1.12.x/src/Type/TypeTraverser.php instead

**Comment by Seldaek** [comment]
Thanks, that solves it for replaceCallbackStrictGroups I think. The problem with isMatchAllStrictGroups and matchAllStrictGroups from #34 is still there tho right?

**Pull request merged**